simulator/kernel/mars/t/unit_test.cpp

```diff
@@ -236,59 +236,69 @@ TEST_CASE( "MARS: open non existing file")
     CHECK( sys.sim->read_cpu_register( v0) == all_ones<uint64>());
 }
 
-TEST_CASE( "MARS: open, write, read and close a file")
+static void write_buff_to_file( System* sys, uint64 descr,
+                                uint64 buff_ptr, uint64 chars_to_write)
+{
+    sys->sim->write_cpu_register( v0, 15u); //write to file
+    sys->sim->write_cpu_register( a0, descr);
+    sys->sim->write_cpu_register( a1, buff_ptr);
+    sys->sim->write_cpu_register( a2, chars_to_write);
+}
+
+static void open_file( System* sys, uint64 filename_ptr, uint64 flags)
+{
+    sys->sim->write_cpu_register( v0, 13u); // open file
+    sys->sim->write_cpu_register( a0, filename_ptr);
+    sys->sim->write_cpu_register( a1, flags);
+    //return sys->sim->read_cpu_register( v0); //FIXME: tests failed
```

Comment:
The test fails since you only updated registers without performing actual system call.
You may either leave the function as is with better name (`prepare_to_open_file`) or put the call inside (`mars_kernel->execute`).